### PR TITLE
Fix Kedro in Azure ML with SQLiteSessionStore 

### PR DESCRIPTION
**Pull request by SajidAlamQB**
## Description

Related to: https://github.com/kedro-org/kedro-viz/issues/1991

This PR addresses issues with SQLite database locking in Azure Machine Learning (AML).

Reading online other users have also had this problem. It stems from the fact that in Azure ML the `/home` directory is mounted as CIFS filesystem which can't deal with SQLite3 lock. Read more here: https://stackoverflow.com/a/54051016

To resolve this, we can enable [Write-Ahead Logging (WAL)](https://sqlite.org/wal.html) mode to fix these locking issues. 

We can either:

1. **Adopt WAL for all environments**: If we go with this option the `session_store.db` file will have a small change in its internal structure to indicate that it is now using WAL mode. This change is stored in the database's header.

2. **Enable WAL mode conditionally for Azure ML:** So we activate WAL mode only when the code detects that it is running in an AML environment. (Currently implemented in this PR).

## Development notes

- Included `PRAGMA journal_mode=WAL;` to enable `WAL` mode.

- Implemented a condition to detect when it is running in an Azure Machine Learning environment using environment variables specific to Azure:`AZUREML_ARM_SUBSCRIPTION`, `AZUREML_ARM_RESOURCEGROUP`, or `AZUREML_RUN_ID`.

Provided two options:

1. Apply WAL mode universally across all environments.
2. Apply WAL mode only when the AML environment is detected, using the `is_azure_ml` flag.

## QA notes

I tested this in Azure Machine Learning studio and confirmed that the database operates in WAL mode without the locking issues.

Tested locally as well it works as expected with and without enabling WAL mode.

image

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: package/kedro_viz/database.py [hunk 1]
```diff
@@ -10,9 +12,27 @@ def make_db_session_factory(session_store_location: str) -> sessionmaker:
     """SQLAlchemy connection to a SQLite DB"""
     database_url = f"sqlite:///{session_store_location}"
     engine = create_engine(database_url, connect_args={"check_same_thread": False})
-    session_class = sessionmaker(engine)
     # TODO: making db session factory shouldn't depend on models.
     # So want to move the table creation elsewhere ideally.
     # But this means returning engine as well as session class.
+
+    # Check if we are running in an Azure ML environment.
+    is_azure_ml = any(
```
Comment (rashidakanchwala) [inline package/kedro_viz/database.py:20]: Looks good :) can we move the azure logic in another function apply_azure_ml_wal_mode ?

Comment (SajidAlamQB) [reply package/kedro_viz/database.py:13]: That makes sense!

**Review by rashidakanchwala — APPROVED** [review]
LGTM. thanks

**Review by jitu5 — APPROVED** [review]

**Review by Huongg — APPROVED** [review]
thanks @SajidAlamQB , LGTM too 👍

**Pull request merged**